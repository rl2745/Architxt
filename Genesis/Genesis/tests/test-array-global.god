void init() {}
void update(int f) {}

int[] x;
string[] y;

int main() {
	int i;
    int j;
    int[] z;

	x = new int[64];
    y = new string[10];
    z = new int[512];

    for (i = 0; i < 64; i = i + 1) {
        x[i] = i;
        for (j = 0; j < 8; j = j + 1) {
            z[j * 64 + i] = j * 64 + i;
        }
    }

    y[8] = "lol";

    for (i = 0; i < 64; i = i + 1) {
        printfl(x[i]);
        print(z[i * 5]);
        prints(y[8]);
    }

	return 0;
}
