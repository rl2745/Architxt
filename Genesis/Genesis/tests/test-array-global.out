0.000000
0
lol
1.000000
5
lol
2.000000
10
lol
3.000000
15
lol
4.000000
20
lol
5.000000
25
lol
6.000000
30
lol
7.000000
35
lol
8.000000
40
lol
9.000000
45
lol
10.000000
50
lol
11.000000
55
lol
12.000000
60
lol
13.000000
65
lol
14.000000
70
lol
15.000000
75
lol
16.000000
80
lol
17.000000
85
lol
18.000000
90
lol
19.000000
95
lol
20.000000
100
lol
21.000000
105
lol
22.000000
110
lol
23.000000
115
lol
24.000000
120
lol
25.000000
125
lol
26.000000
130
lol
27.000000
135
lol
28.000000
140
lol
29.000000
145
lol
30.000000
150
lol
31.000000
155
lol
32.000000
160
lol
33.000000
165
lol
34.000000
170
lol
35.000000
175
lol
36.000000
180
lol
37.000000
185
lol
38.000000
190
lol
39.000000
195
lol
40.000000
200
lol
41.000000
205
lol
42.000000
210
lol
43.000000
215
lol
44.000000
220
lol
45.000000
225
lol
46.000000
230
lol
47.000000
235
lol
48.000000
240
lol
49.000000
245
lol
50.000000
250
lol
51.000000
255
lol
52.000000
260
lol
53.000000
265
lol
54.000000
270
lol
55.000000
275
lol
56.000000
280
lol
57.000000
285
lol
58.000000
290
lol
59.000000
295
lol
60.000000
300
lol
61.000000
305
lol
62.000000
310
lol
63.000000
315
lol
